src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ErodeDilateCoordinationNumber.cpp

```diff
@@ -0,0 +1,211 @@
+
+  const std::string attrMatName = m_InputValues->FeatureIdsArrayPath.getTargetName();
+  const std::vector<std::shared_ptr<IDataArray>> voxelArrays = complex::GenerateDataArrayList(m_DataStructure, m_InputValues->FeatureIdsArrayPath, m_InputValues->IgnoredDataArrayPaths);
+
+  std::vector<int32> featureCount(numFeatures + 1, 0);
+  std::vector<int32> coordinationNumber(totalPoints, 0);
+  bool keepGoing = true;
+  int32 counter = 1;
+
+  while(counter > 0 && keepGoing)
+  {
+    counter = 0;
+    if(!m_InputValues->Loop)
+    {
+      keepGoing = false;
+    }
+
+    for(int64 zIndex = 0; zIndex < dims[2]; zIndex++)
+    {
+      const int64 zStride = dims[0] * dims[1] * zIndex;
+      for(int64 yIndex = 0; yIndex < dims[1]; yIndex++)
+      {
+        const int64 yStride = dims[0] * yIndex;
+        for(int64 xIndex = 0; xIndex < dims[0]; xIndex++)
+        {
+          const int64 voxelIndex = zStride + yStride + xIndex;
+          const int32 featureName = featureIds[voxelIndex];
+          int32 coordination = 0;
+          int32 most = 0;
+          for(int32 neighPointIdx = 0; neighPointIdx < 6; neighPointIdx++)
+          {
+            const int64 neighborPoint = voxelIndex + neighpoints[neighPointIdx];
+            if(neighPointIdx == 0 && zIndex == 0)
+            {
+              continue;
+            }
+            if(neighPointIdx == 5 && zIndex == (dims[2] - 1))
+            {
+              continue;
+            }
+            if(neighPointIdx == 1 && yIndex == 0)
+            {
+              continue;
+            }
+            if(neighPointIdx == 4 && yIndex == (dims[1] - 1))
+            {
+              continue;
+            }
+            if(neighPointIdx == 2 && xIndex == 0)
+            {
+              continue;
+            }
+            if(neighPointIdx == 3 && xIndex == (dims[0] - 1))
+            {
+              continue;
+            }
+
+            const int32 feature = featureIds[neighborPoint];
+            if((featureName > 0 && feature == 0) || (featureName == 0 && feature > 0))
+            {
+              coordination = coordination + 1;
+              featureCount[feature]++;
+              const int32 current = featureCount[feature];
+              if(current > most)
+              {
+                most = current;
+                neighbors[voxelIndex] = neighborPoint;
+              }
+            }
+          }
+          coordinationNumber[voxelIndex] = coordination;
+          const int64 neighbor = neighbors[voxelIndex];
+          if(coordinationNumber[voxelIndex] >= m_InputValues->CoordinationNumber && coordinationNumber[voxelIndex] > 0)
+          {
+            /******************************************************************
+             * If this section is slow it is because we are having to use the
+             * ExecuteDataFunction<T>() in order to call "copyTuple()" because
+             * "copyTuple()" isn't in the IArray API set. Oh well.
+             */
```

Comment:
Is there any way we can utilize `CopyTupleUsingIndexList` from DataArrayUtilities? If so we could parallelize this speeding it up significantly.

Reply:
That algorithm takes way too much memory to perform the same copying. I'll leave this in here until we find a minimally memory usage algorithm.